### PR TITLE
[#155901247] PinScreen

### PR DESCRIPTION
**Pull request by fpersico**
Add the PinScreen that allow the user to set the PIN.

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Remember to fix the PR title by removing WIP wording when ready
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#155901247](https://www.pivotaltracker.com/story/show/155901247): Sviluppare lo Screen di inserimento e conferma PIN

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: js/screens/onboarding/PinScreen.js [hunk 1]
```diff
@@ -291,6 +294,7 @@ class PinScreen extends React.Component<Props, State> {
 }
 
 const mapStateToProps = (state: GlobalState): ReduxMappedProps => ({
+  // Check from the store is there was an error creating the PIN (saving into the Keystore)
```
Comment (cloudify) [inline js/screens/onboarding/PinScreen.js]: > Checks from the store whether there was an error while creating the PIN (e.g. saving into the Keystore)

File: js/sagas/onboarding.js [hunk 2]
```diff
@@ -44,8 +50,52 @@ function* pinCheckSaga(): Saga<void> {
     })
     yield put(navigateToOnboardingPinScreenAction)
 
-    // Here we wait the user to create a PIN
-    yield take(PIN_CREATE_REQUEST)
+    let isPinSaved = false
+
+    // Loop until PIN successfully saved in the Keystore
+    while (!isPinSaved) {
```
Comment (gunzip) [inline js/sagas/onboarding.js]: is there any way to cancel the action ?

File: js/screens/onboarding/PinScreen.js [hunk 3]
```diff
@@ -6,31 +6,296 @@ import {
...
+              (this.pinConfirmComponent = el)
+            }
+            secureTextEntry
+            keyboardType="numeric"
+            autoFocus
+            compareWithCode={pinState.pin}
+            className="border-b"
+            codeLength={5}
+            cellBorderWidth={2}
+            inactiveColor={'rgba(92,111,130,0.5)'}
```
Comment (gunzip) [inline js/screens/onboarding/PinScreen.js]: can we take these colors from a "palette" ? (a list / array of constants that represents color, ie. "brandColor: rgba..."

File: js/screens/onboarding/PinScreen.js [hunk 4]
```diff
@@ -6,31 +6,296 @@ import {
...
+            activeColor={'rgba(23,50,77,1)'}
+            onFulfill={this.onPinConfirmFulfill}
+            codeInputStyle={{ fontSize: 24, height: 56 }}
```
Comment (gunzip) [inline js/screens/onboarding/PinScreen.js]: with another PR, can we have these fontSize taken from a predefined set of constants ? alike
https://github.com/jxnblk/styled-system

File: js/sagas/onboarding.js (hunk 2 above)
Comment (fpersico) [reply js/sagas/onboarding.js]: If you mean cancel explictly, we could add a race [PIN_CREATE_REQUEST, PIN_CREATE_CANCEL] than if PIN_CREATE_CANCEL Win in some way we exit from the loop. The problem is that the user MUST complete the PinStep. Which use case are you thinking of?

File: js/screens/onboarding/PinScreen.js (hunk 3 above)
Comment (fpersico) [reply js/screens/onboarding/PinScreen.js]: You're right I forgot to change them.

File: js/screens/onboarding/PinScreen.js (hunk 4 above)
Comment (fpersico) [reply js/screens/onboarding/PinScreen.js]: I think we can change this also in this PR.

File: js/components/ui/TextWithIcon.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import { connectStyle } from 'native-base-shoutem-theme'
+
+type Props = ViewProps
+
+/**
+ * A simple component to display an Icon and a Text side-by-side
+ */
+class TextWithIcon extends React.Component<Props> {
+  render(): React.Node {
+    return <View {...this.props} />
```
Comment (cloudify) [inline js/components/ui/TextWithIcon.js]: is this just a View?

File: js/sagas/onboarding.js (hunk 2 above)
Comment (fpersico) [reply js/sagas/onboarding.js]: If the user is in the PinScreen it is already LoggedIn and must complete the Onboarding. In which case(s) we must break this Loop?

File: js/screens/onboarding/BiometricScreen.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+type OwnProps = {
+  navigation: NavigationScreenProp<NavigationState>
+}
+
+type Props = ReduxMappedProps & ReduxProps & OwnProps
+
+/**
+ * A screen that allows the user to set the Biometric.
+ *
```
[the remark this answers is not in the document]
Comment (fpersico) [reply js/screens/onboarding/BiometricScreen.js]: Done. https://www.pivotaltracker.com/story/show/156990825

**Comment by cloudify** [comment]
Converted to TS see #100

**Pull request closed without merging**